### PR TITLE
Add more specs for YIP-0001

### PR DESCRIPTION
**Pull request by mnaichuk**

### CODE REVIEW

File: docs/YIP/YIP-0001-price-precision.md [hunk 1]
```diff
@@ -17,13 +17,18 @@ In the context of YN, we must have a network global configuration to make member
...
 
 The precision level of all trading prices is calculated based on significant figures. Yellow members uses seven (7) significant digits for the prices of all pairs of the network.
+Alternative options for significant figures is in range of 5 - 8.
 
-Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.
+Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340 those examples for 5 significant figures.
 
 This is similar to how traditional global markets handle the precision of small, medium, and large values. The reasoning behind this is that the number of decimals becomes less important as the amount increases. The opposite is true for minimal amounts, where greater precision is more valuable.
 
+Amount and price precision allows up to 8 decimals. Anything exceeding this will be rounded to the 8th decimal.
```
Comment (mnaichuk) [inline docs/YIP/YIP-0001-price-precision.md:26]: price - 10 decimal precision, 5 significant figures

File: docs/YIP/YIP-0001-price-precision.md [hunk 2]
```diff
@@ -17,13 +17,18 @@ In the context of YN, we must have a network global configuration to make member
...
+
 ## Consequences
 
 1. All API outputting prices should be truncated and formation using seven (7) significant figures
-2. All API input must reject orders where prices are lower in precision
-3. All frontend and UI clients should truncate order price input to seven (7) significant figures
+2. All API outputting prices and amounts allows up to truncated 8 decimals. Anything exceeding this will be rounded to the 8th decimal.
```
Comment (mnaichuk) [inline docs/YIP/YIP-0001-price-precision.md:31]: API level - return error
UI - will not allow to input

File: docs/YIP/YIP-0001-price-precision.md [hunk 3]
```diff
@@ -17,13 +17,18 @@ In the context of YN, we must have a network global configuration to make member
 Significant figures, also known as the precision of a number in positional notation, are digits in the number that are reliable and necessary to indicate the quantity of something.
 
 The precision level of all trading prices is calculated based on significant figures. Yellow members uses seven (7) significant digits for the prices of all pairs of the network.
+Alternative options for significant figures is in range of 5 - 8.
 
-Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.
+Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340 those examples for 5 significant figures.
```
Comment (nksazonov) [inline docs/YIP/YIP-0001-price-precision.md:22]: 'five significant digits' is repeated twice

File: docs/YIP/YIP-0001-price-precision.md [hunk 4]
```diff
@@ -14,16 +14,36 @@ In the context of YN, we must have a network global configuration to make member
 
 ## Decision
 
+### Definitions
+
+**Price** - asset price in quote currency.
+
+**Quantity** - in trading, quantity usually refers to the number of units or shares
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:21]: The second `quantity` is excessive

File: docs/YIP/YIP-0001-price-precision.md [hunk 5]
```diff
@@ -14,16 +14,36 @@ In the context of YN, we must have a network global configuration to make member
...
+of a particular security or asset that is being bought or sold.
+For instance, if an investor buys 100 shares of a company's stock, the quantity in this case is 100.
+
+**Value** - stands for USD market value, value is always in USD (for example for market BTC/ETH value will be in USD).
+
+| Number      | Round to |
+| ----------- | ----------- |
+| Price       | 18       |
+| Quantity    | 8        |
+| Value       | 2        |
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:31]: Add an explanation about what this table is.

File: docs/YIP/YIP-0001-price-precision.md [hunk 6]
```diff
@@ -14,16 +14,36 @@ In the context of YN, we must have a network global configuration to make member
...
+The precision level of all trading prices is calculated based on significant figures. Yellow members uses five (5) significant digits for the prices of all pairs of the network.
+Alternative options for significant figures is in range of 5 - 8.
 
 Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.
 
-This is similar to how traditional global markets handle the precision of small, medium, and large values. The reasoning behind this is that the number of decimals becomes less important as the amount increases. The opposite is true for minimal amounts, where greater precision is more valuable.
+This is similar to how traditional global markets handle the precision of small, medium, and large values. The reasoning behind this is that the number of decimals becomes less important as the quantity increases. The opposite is true for minimal amounts, where greater precision is more valuable.
+
+Quantity precision allows up to 8 decimals. Anything exceeding this will be rounded to the 8th decimal.
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:42]: Change `will` to `shall` according to meta YIP

File: docs/YIP/YIP-0001-price-precision.md [hunk 7]
```diff
@@ -14,16 +14,36 @@ In the context of YN, we must have a network global configuration to make member
...
 
 ## Consequences
 
-1. All API outputting prices should be truncated and formation using seven (7) significant figures
-2. All API input must reject orders where prices are lower in precision
-3. All frontend and UI clients should truncate order price input to seven (7) significant figures
+1. All API outputting prices should be truncated and formation using five (5) significant figures
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:46]: Change  `should` to `shall` according to meta YIP

File: docs/YIP/YIP-0001-price-precision.md [hunk 8]
```diff
@@ -14,16 +14,36 @@ In the context of YN, we must have a network global configuration to make member
...
+3. All API input will return an error for orders where prices are lower in precision
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:47]: `shall` again. 
Is it possible to define the error format and message at the YIP stage?

File: docs/YIP/YIP-0001-price-precision.md [hunk 9]
```diff
@@ -14,16 +14,36 @@ In the context of YN, we must have a network global configuration to make member
...
+4. All frontend and UI clients should truncate order price input to five (5) significant figures
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:48]: Change `should` to `shall` according to meta YIP

File: docs/YIP/YIP-0001-price-precision.md [hunk 10]
```diff
@@ -14,16 +14,36 @@ In the context of YN, we must have a network global configuration to make member
...
+5. Backend and UI pkgs should support easy change of significant figures
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:49]: Change `should` to `shall` according to meta YIP.
Define the limits of change, like from 5 to 8 including 5 and 8.

File: docs/YIP/YIP-0001-price-precision.md [hunk 11]
```diff
@@ -14,16 +14,36 @@ In the context of YN, we must have a network global configuration to make member
...
+Alternative options for significant figures is in range of 5 - 8.
 
 Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.
 
-This is similar to how traditional global markets handle the precision of small, medium, and large values. The reasoning behind this is that the number of decimals becomes less important as the amount increases. The opposite is true for minimal amounts, where greater precision is more valuable.
+This is similar to how traditional global markets handle the precision of small, medium, and large values. The reasoning behind this is that the number of decimals becomes less important as the quantity increases. The opposite is true for minimal amounts, where greater precision is more valuable.
+
+Quantity precision allows up to 8 decimals. Anything exceeding this will be rounded to the 8th decimal.
 
 ## Consequences
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:44]: What is consensus for other parameters like Quantity and Value?

File: docs/YIP/YIP-0001-price-precision.md (hunk 5 above)
Comment (mnaichuk) [reply docs/YIP/YIP-0001-price-precision.md]: we dropped value at this point

File: docs/YIP/YIP-0001-price-precision.md (hunk 6 above)
Comment (mnaichuk) [reply docs/YIP/YIP-0001-price-precision.md]: fixed

File: docs/YIP/YIP-0001-price-precision.md (hunk 8 above)
Comment (mnaichuk) [reply docs/YIP/YIP-0001-price-precision.md]: I'm not ready to define final error right now

File: docs/YIP/YIP-0001-price-precision.md (hunk 11 above)
Comment (mnaichuk) [reply docs/YIP/YIP-0001-price-precision.md:41]: we didn't have consensus on value param and we don't need it at this point

File: docs/YIP/YIP-0001-price-precision.md [hunk 12]
```diff
@@ -14,16 +14,33 @@ In the context of YN, we must have a network global configuration to make member
...
+**Quantity** - in trading, usually refers to the number of units or shares
+of a particular security or asset that is being bought or sold.
+For instance, if an investor buys 100 shares of a company's stock, the quantity in this case is 100.
+
+| Number      | Round to |
+| ----------- | ----------- |
+| Price       | 18       |
+| Quantity    | 8        |
+
 Significant figures, also known as the precision of a number in positional notation, are digits in the number that are reliable and necessary to indicate the quantity of something.
```
Comment (nksazonov) [inline docs/YIP/YIP-0001-price-precision.md:30]:
```suggestion
**Significant figures** (also known as the precision of a number in positional notation) - are digits in the number that are reliable and necessary to indicate the quantity of something.
```

File: docs/YIP/YIP-0001-price-precision.md [hunk 13]
```diff
@@ -14,16 +14,33 @@ In the context of YN, we must have a network global configuration to make member
...
 
-The precision level of all trading prices is calculated based on significant figures. Yellow members uses seven (7) significant digits for the prices of all pairs of the network.
+The precision level of all trading prices is calculated based on significant figures. Yellow members uses five (5) significant digits for the prices of all pairs of the network.
+Alternative options for significant figures is in range of 5 - 8.
```
Comment (nksazonov) [inline docs/YIP/YIP-0001-price-precision.md:33]:
```suggestion
Alternative options for significant figures are in range of 5 - 8.
```

File: docs/YIP/YIP-0001-price-precision.md [hunk 14]
```diff
@@ -14,16 +14,33 @@ In the context of YN, we must have a network global configuration to make member
...
 
 Some examples of five significant digits are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.
```
Comment (nksazonov) [inline docs/YIP/YIP-0001-price-precision.md:35]:
```suggestion
Some examples of _five significant digits_ are 1.0234, 10.234, 120.34, 1234.5, 0.012345, and 0.00012340.
```

File: docs/YIP/YIP-0001-price-precision.md [hunk 15]
```diff
@@ -14,16 +14,33 @@ In the context of YN, we must have a network global configuration to make member
...
+
+| Number      | Round to |
+| ----------- | ----------- |
+| Price       | 18       |
+| Quantity    | 8        |
+
 Significant figures, also known as the precision of a number in positional notation, are digits in the number that are reliable and necessary to indicate the quantity of something.
 
-The precision level of all trading prices is calculated based on significant figures. Yellow members uses seven (7) significant digits for the prices of all pairs of the network.
+The precision level of all trading prices is calculated based on significant figures. Yellow members uses five (5) significant digits for the prices of all pairs of the network.
```
Comment (nksazonov) [inline docs/YIP/YIP-0001-price-precision.md:32]:
```suggestion
The precision level of all trading prices is calculated based on significant figures. Yellow members shall use five (5) significant digits for the prices of all pairs of the network.
```

**Review by nksazonov — CHANGES_REQUESTED** [review]

**Review by nksazonov — APPROVED** [review]

File: docs/YIP/YIP-0001-price-precision.md [hunk 16]
```diff
@@ -14,16 +14,33 @@ In the context of YN, we must have a network global configuration to make member
...
+
+Quantity precision allows up to 8 decimals. Anything exceeding this shall be rounded to the 8th decimal.
 
 ## Consequences
 
-1. All API outputting prices should be truncated and formation using seven (7) significant figures
-2. All API input must reject orders where prices are lower in precision
-3. All frontend and UI clients should truncate order price input to seven (7) significant figures
+1. All API outputting prices shall be truncated and formation using five (5) significant figures
+3. All API input shall return an error for orders where prices or amount are lower in precision
```
Comment (pkucherenk0) [inline docs/YIP/YIP-0001-price-precision.md:44]: All API input shall return an error for orders where prices or amounts exceed 18 and 8 precision respectfully

**Review by akushniruk — APPROVED** [review]

**Review by pkucherenk0 — APPROVED** [review]

**Pull request merged**